var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -187,33 +187,33 @@ def url_for_version(self, version):
         return url.format(version.up_to(2), version)
 
     def flag_handler(self, name, flags):
+        spec = self.spec
         cmake_flags = []
 
         if name == "cflags":
-            if self.spec.satisfies('%gcc') \
-                    or self.spec.satisfies('%clang'):
+            if spec.compiler.name in ['gcc', 'clang', 'apple-clang']:
```

Comment:
Is there a compiler that won't need this flag changed? Seems that this will be an increasing list as other compilers are used?

Reply:
I think idea is that this is a whitelist for compilers that are known to support the flag. It's bad to introduce flags that might break other compilers.